db/db_impl/db_impl_write.cc

```diff
@@ -220,7 +220,8 @@ Status DBImpl::WriteImpl(const WriteOptions& write_options,
 
   bool need_log_sync = write_options.sync;
   bool need_log_dir_sync = need_log_sync && !log_dir_synced_;
-  if (!two_write_queues_ || !disable_memtable) {
+  assert(!two_write_queues_ || !disable_memtable);
```

Comment:
Should `two_write_queues_ && disable_memtable` is true, we would have returned from line 112:
```
if (two_write_queues_ && disable_memtable) {
  return WriteImplWALOnly(...);
}
```